File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program20.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept number from user and return difference between summation of even digits and summation of odd digits.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.*;

class Solution
{
	public int iValue;

	public Solution(int iValue)
	{
		this.iValue = iValue;
	}

	public int CountDiff()
	{
                      	
		int iEven=0,iOdd=0;
		while(iValue!=0)
		{
			int iDigit = iValue%10;
			if(iDigit%2==0)
			{
				iEven=iEven+iDigit;
			}
			else
			{
				iOdd=iOdd+iDigit;
			}
			iValue = iValue/10;
		}        
		return iEven-iOdd;
	
	}
}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		int iRet=0;
		System.out.println("Enter number: ");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iNo);
		iRet = obj.CountDiff();
	
		System.out.println("Difference is: "+iRet);
	}
}

/*
OUTPUT:

Enter number:
2395
Difference is: -15

*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program20.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept number from user and return difference between summation of even digits and summation of odd digits.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Display
{
	private:
		int iNo;

	public:

		Display(int iNo)
		{
			this->iNo = iNo;
		}	

		int CountDiff()
		{
			int iDigit = 0,iSumEven=0,iSumOdd=0,iAns=0;

			if(iNo<0)
			{
				iNo = -iNo;
			}

			while(iNo>0)
			{
				iDigit = iNo%10;
				if(iDigit%2==0)
				{
					iSumEven = iSumEven+iDigit;
				}
				else
				{
					iSumOdd = iSumOdd+iDigit;
				}
				iNo = iNo/10;
			}

			iAns = iSumEven-iSumOdd;
			return iAns;
		}
};

int main()
{
	int iValue=0,iRet=0;

	cout<<"Enter Number: ";
	cin>>iValue;

	Display *dObj = new Display(iValue);
	iRet = dObj->CountDiff();

	cout<<"Difference is: "<<iRet<<"\n";
	return 0;
}

/*
OUTPUT:

Enter Number: 2395
Difference is: -15

*/
